var/spack/repos/builtin/packages/rocm-config/package.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
+
+import itertools
+import pathlib
+
+from spack.package import *
+
+
+# ask the driver what gpus are installed
+def _query_kfd_gfx_targets(*known_targets, multi=False):
+    targets = []
+    try:
+        nodes = pathlib.Path("/sys/class/kfd/kfd/topology/nodes")
+        for filename in nodes.glob("*/properties"):
+            with filename.open(encoding="utf-8") as f:
+                for line in f:
+                    label = "gfx_target_version "
+                    if not line.startswith(label):
+                        continue
+                    version = int(line[len(label) :])
+                    if not version:
+                        break
+                    major_version = version // 10000
+                    minor_version = (version // 100) % 100
+                    step_version = version % 100
+                    target = "gfx{:d}{:x}{:x}".format(major_version, minor_version, step_version)
+                    if target in known_targets:
+                        targets.append(target)
+                    break
+    except (OSError, ValueError):
+        pass  # query is best-effort
+    if targets:
+        if multi:
+            return targets
+        else:
+            return targets[-1]
+    else:
+        return "none"
+
+def target_strip_feature(target):
+    return target.split(":")[0]
+
+class RocmConfig(BundlePackage):
+    """Target and Version configurations for ROCm Stacks"""
+
+    # ROCm stack versions available in Spack
+    rocm_versions = {
+        "5.4.3":  {"preferred": True},
```

Comment:
No rocm version >= 6 ?

Reply:
Thanks for the feed back, none of the versions you mentioned existed when this work was originally done. This was just rebased and draft PR'd to get feedback on modeling ROCm packages like this at the moment.